org.eclipse.m2e.core/src/org/eclipse/m2e/core/project/configurator/AbstractProjectConfigurator.java

```diff
@@ -154,28 +162,56 @@ public void setInitializationData(IConfigurationElement config, String propertyN
     this.name = config.getAttribute(ATTR_NAME);
   }
 
-  // TODO move to a helper
+  // TODO move to a helper, to which?
   public static void addNature(IProject project, String natureId, IProgressMonitor monitor) throws CoreException {
     addNature(project, natureId, IResource.KEEP_HISTORY, monitor);
   }
 
   /**
    * @since 1.3
    */
-  // TODO move to a helper
+  // TODO move to a helper, to which?
   public static void addNature(IProject project, String natureId, int updateFlags, IProgressMonitor monitor)
       throws CoreException {
     if(!project.hasNature(natureId)) {
       IProjectDescription description = project.getDescription();
-      String[] prevNatures = description.getNatureIds();
-      String[] newNatures = new String[prevNatures.length + 1];
-      System.arraycopy(prevNatures, 0, newNatures, 1, prevNatures.length);
-      newNatures[0] = natureId;
-      description.setNatureIds(newNatures);
+      var natures = Stream.concat(Stream.of(natureId), Arrays.stream(description.getNatureIds()));
+      description.setNatureIds(natures.toArray(String[]::new));
       project.setDescription(description, updateFlags, monitor);
     }
   }
 
+  /**
+   * Creates a problem marker in the pom.xml file at the specified attribute of the given execution with the passed
+   * message and severity.
+   * <p>
+   * If the attribute of the execution is specified in a parent pom.xml outside of the workspace the marker is created
+   * at the parent-element of the request's project.
+   * </p>
+   * 
+   * @param execution the execution
+   * @param attribute the attribute to mark
```

Comment:
It would be good to elaborate which values are valid here

Reply:
I assume that you can use any attribute of the plug-ins XML configuration, but I can check that.

Reply:
Maven pom.xml doesn’t use XML attributes. I have seen that element names have been given here.

Reply:
There are at least `combine.*` attribute that Maven uses: https://maven.apache.org/pom.html#advanced_configuration_inheritance

Reply:
Those are not meant here IIUC.

Reply:
Actually (XML) `elements` are meant her, not `attributes`. Sorry for the confusion.
I just blindly copied that from the internal method.